### ISSUE TITLE
Capture file responded by a form POST

### ISSUE THREAD

**Issue by borisschapira** [segmentio/nightmare#422]
Hello,

I have a form that respond to submission with a CSV file. Do you know how I could test it ?

* test that a file in responded ?
* test the file name ?
* test the file size ?

**Comment by rosshinkley** [comment]
TLDR: You currently cannot test downloads this way.  There are plans to address this issue.

A slightly longer explanation: use-cases for how Electron handles downloads - the `will-download` event - are very different.  PR #382 attempts to manage downloads, but it was decided that downloads should be handled in one or more plugins (see #391) - one for handling them in parallel, and possibly another for handling them inside of the Nightmare pipeline.  It may also be worth reading #332, #371, and #151 for more background.

**Comment by borisschapira** [comment]
Thanks a lot for this detailed explanation that/certainly took time to
write. It's much appreciated !

**Comment by rosshinkley** [comment]
With the release of #425 in 2.3.0, I've gone ahead and released plugins that will take care of downloads:

* [nightmare-download-manager](https://github.com/rosshinkley/nightmare-download-manager)
* [nightmare-inline-download](https://github.com/rosshinkley/nightmare-inline-download)

I'm considering this issue resolved.  If you have further questions or problems, please open a new issue.

**Comment by borisschapira** [comment]
Thanks a lot, @rosshinkley, what a great bunch of plugins, that's awesome!
